### PR TITLE
REPO-1742: Add relativePath property to optional "path" object used for nodes

### PR DESCRIPTION
**Pull request by anechifor**
…or nodes

### CODE REVIEW

File: src/main/java/org/alfresco/rest/api/impl/NodesImpl.java [hunk 1]
```diff
@@ -1311,8 +1311,46 @@ public Node get(int index)
...
+                            StringBuilder sb = new StringBuilder(pathInfos.size() * 20);
+                            for (String fileInfo : pathInfos)
+                            {
+                                sb.append("/");
+                                sb.append(fileInfo);
+                            }
+
+                            node.getPath().setRelativePath(sb.toString());
+                        }
+
```
Comment (derekhulley) [inline src/main/java/org/alfresco/rest/api/impl/NodesImpl.java]: Screen-eating whitespace

File: src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java [hunk 2]
```diff
@@ -4343,6 +4343,54 @@ public void testUpdateNodePermissions() throws Exception
             clearAuthorityContext();
         }
     }
+    
+    /**
+     * Test upload using relativePath
+     * @throws Exception
```
Comment (derekhulley) [inline src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java]: Remove empty attributes from comments; they don't add value.

File: src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java [hunk 3]
```diff
@@ -4343,6 +4343,54 @@ public void testUpdateNodePermissions() throws Exception
...
         }
     }
+    
+    /**
+     * Test upload using relativePath
+     * @throws Exception 
+     * 
+     */
+    @Test
+    public void testUploadUsingRelativePath() throws Exception{
```
Comment (derekhulley) [inline src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java]: Formatting

File: src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java [hunk 4]
```diff
@@ -4343,6 +4343,54 @@ public void testUpdateNodePermissions() throws Exception
             clearAuthorityContext();
         }
     }
+    
+    /**
+     * Test upload using relativePath
+     * @throws Exception 
+     *
```
Comment (derekhulley) [inline src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java]: Unnecessary whitespace

File: src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java [hunk 5]
```diff
@@ -4773,5 +4821,7 @@ public String getScope()
     {
         return "public";
     }
+
```
Comment (derekhulley) [inline src/test/java/org/alfresco/rest/api/tests/NodeApiTest.java]: Unnecessary whitespace

File: src/main/java/org/alfresco/rest/api/impl/NodesImpl.java [hunk 6]
```diff
@@ -1311,8 +1311,48 @@ public Node get(int index)
...
+                            sb.append(fileInfo);
+                        }
+
+                        node.getPath().setRelativePath(sb.toString());
+                    }
+
+                }
+
+                catch (FileNotFoundException e)
+                {
```
Comment (cristianturlica) [inline src/main/java/org/alfresco/rest/api/impl/NodesImpl.java]: Please add a comment if this is expected.
